test/integration/router_container_test.rb

```diff
@@ -13,4 +13,17 @@
     response = @app.get('/back/home', lint: true)
     response.body.must_equal 'back'
   end
+
+  it 'matches against host' do
+    @router_container = Hanami::Router.new(scheme: 'https', host: 'hanami.test', port: 443) do
+      mount Front::App, at: '/front', host: 'www.hanami.test'
+      mount Back::App, at: '/front'
+    end
+
+    @app = Rack::MockRequest.new(@router_container)
+    response = @app.get('https://www.hanami.test/front/home', lint: true)
+    response.body.must_equal 'front'
+    response = @app.get('/front/home', lint: true)
```

Comment:
@nicolaracco Shall we test this like this? `@app.get('https:/hanami.test/front/home', lint: true)`